File: tic-tac-toe-javafx-java/src/main/java/com/mj/tic/tac/toe/javafx/java/util/Mark.java
package com.mj.tic.tac.toe.javafx.java.util;

import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * Represents a player's mark placed on a Tic-Tac-Toe board cell, consisting
 * of a {@code char} symbol ({@code 'X'} or {@code 'O'}) and the
 * {@link Coordinates} of the cell where it was placed.
 * <p>
 * Instances are created via {@link #Mark(char, Coordinates)} and are
 * immutable, making them safe to use as map keys or in collections.
 *
 * @author Montaser Jamal
 * @version 1.0
 * @email montaser.jjs@gmail.com
 * @phone +962-786258874
 * @since 22-01-2023
 */

@Data
@AllArgsConstructor
public final class Mark {

    private final char value;

    private final Coordinates coordinates;
}

File: tic-tac-toe-javafx-kotlin/src/main/kotlin/com/mj/tic/tac/toe/javafx/kotlin/util/Mark.kt
package com.mj.tic.tac.toe.javafx.kotlin.util

/**
 * Represents a single mark placed on the game board during a move.
 *
 * This data class pairs the visual character displayed on the UI button
 * ('X' or 'O') with the board [Coordinates] where the mark was placed.
 * It is returned by [MarkCellTask] as the result of a successful move
 * and consumed by [PlayAreaPanelController.onTaskSucceeded] to determine
 * the next game state.
 *
 * @property symbol The character representing the mark on the UI
 *   ('X' for the first player, 'O' for the second).
 * @property coordinates The zero-based board position of this mark.
 * @author Montaser Jamal
 * @version 1.0
 * @email montaser.jjs@gmail.com
 * @phone +962-786258874
 * @since 25-01-2023
 */

data class Mark(val symbol: Char, val coordinates: Coordinates)
